### PR TITLE
Add Installer TechPreview presubmit jobs

### PR DESCRIPTION
**Pull request by sadasu**

### CODE REVIEW

File: ci-operator/config/openshift/installer/openshift-installer-master.yaml [hunk 1]
```diff
@@ -404,6 +404,14 @@ tests:
     cluster_profile: azure4
     workflow: openshift-e2e-azure
   timeout: 6h0m0s
+- as: e2e-azure-ovn-techpreview
+  run_if_changed: azure
```
Comment (r4f4) [inline ci-operator/config/openshift/installer/openshift-installer-master.yaml:408]:
```suggestion
  optional: true
```
We don't want this job being automatically triggered.

File: ci-operator/config/openshift/installer/openshift-installer-master__altinfra.yaml [hunk 2]
```diff
@@ -243,6 +243,15 @@ tests:
...
   timeout: 6h0m0s
+- always_run: false
+  as: e2e-azure-capi-ovn-techpreview
+  optional: true
+  steps:
+    cluster_profile: azure4
+    env:
+      FEATURE_SET: TechPreviewNoUpgrade
+    workflow: openshift-e2e-azure
+  timeout: 6h0m0s
```
Comment (r4f4) [inline ci-operator/config/openshift/installer/openshift-installer-master__altinfra.yaml:254]:
```suggestion
```
We don't need this in the altinfra variant.

File: ci-operator/config/openshift/installer/openshift-installer-master.yaml (hunk 1 above)
Comment (patrickdillon) [reply ci-operator/config/openshift/installer/openshift-installer-master.yaml:408]: also `always_run: false`

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml [hunk 3]
```diff
@@ -5138,6 +5138,87 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-azure-ovn-shared-vpc,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (r4f4) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml:5142]: This option sticks between `make update` calls, so you'll have to edit it manually.

**Comment by r4f4** [comment]
They should run but just in case:

/pj-rehearse

**Comment by sadasu** [comment]
/pj-rehearse ci/rehearse/openshift/installer/master/e2e-gcp-ovn-techpreview

**Comment by r4f4** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-gcp-ovn-techpreview

**Comment by sadasu** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-aws-ovn-techpreview

**Comment by r4f4** [comment]
Rehearsals passed the install phase and I see the TechPreview set in the install-config.yaml.
Will LGTM when it's not WIP anymore.

**Pull request merged**